SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosSchedulerClient.java

```diff
@@ -173,7 +180,8 @@ private void connect(URI mesosMasterURI, FrameworkInfo frameworkInfo, Singularit
 
     subscribe.processStream(unicastEvents -> {
 
-      final Observable<Event> events = unicastEvents.share();
+      final Observable<Event> events = unicastEvents.share()
+          .subscribeOn(subscribeScheduler);
```

Comment:
Is there any risk in having events pertaining to the same Mesos host end up on two different subscriber threads simultaneously?

Reply:
I still need to stare at it more, but I don’t think so. Most of SchedulerImpl was written assuming it was already concurrent. Offer related traffic is behind a lock, as are any updates to state. Lost executors are mostly just logging. Status updates are a sync and behind request locks. Think that’s all?